go/test/endtoend/onlineddl/vrepl_stress/onlineddl_vrepl_mini_stress_test.go

```diff
@@ -157,6 +160,18 @@ func nextOpOrder() int64 {
 	return opOrder
 }
 
+func TestInitialSetup(t *testing.T) {
+	repo, ok := os.LookupEnv("GITHUB_REPOSITORY") // `ok` tells us the env variable exists, hence that we are running in GitHub CI.
+	t.Logf("==== repo=%v", repo)
+	if ok && repo != "vitessio/vitess" {
+		// `vitessio/vitess` repository enjoys faster runners. Otherwise, GitHub CI has much slower runners
+		// and we have to reduce the workload
+		maxConcurrency = maxConcurrency / 2
+		singleConnectionSleepInterval = singleConnectionSleepInterval * 2
+	}
```

Comment:
The details of runners are dynamic for a repo. IMO we should instead base the settings on the runtime. For example, something like this:
```
	t.Helper()
	vCPUs := runtime.NumCPU()
	maxConcurrency = vCPUs
	sleepModifier := int(math.Min(float64(4-(vCPUs/4)), 1))
	singleConnectionSleepInterval = time.Duration(int(singleConnectionSleepInterval.Milliseconds()) * sleepModifier)
```

Reply:
The runners available for Actions in a given repo can be changed at any time in Settings->Actions->Runners. And the specific runner used for a workflow can be changed at any time in the yaml. That's why making assumptions based on repo isn't a great solution IMO, since what we really care about (and what we're assuming here based on the repo) are the resources/specs for the runner. Noisy neighbors is an unrelated issue (although a real one with VMs generally).